routes/user.js

```diff
@@ -9,48 +10,80 @@ var router = express.Router();
-            res.send(results);
-        });
+        var userId = res.locals.session ? res.locals.session.user : null;
+        permissions.check(userId, {
+            type: "USER_MANAGE",
+            value: userId
+        }).then(function (result) {
+            if (!result) res.sendStatus(403);
+            User.findAll({
+                attributes: ["id", "displayName", "email", "isAdmin"],
+                order: [
+                    ["id", "ASC"]
+                ]
+            }).then(function (results) {
+                res.send(results);
+            });
+        }).done();
     })
     .post(function (req, res, next) {
-        log.info({req: req.body}, "INITIAL REQUEST");
-        if (!res.locals.session) {
-            return res.sendStatus(401);
-        }
+        var makeLink = function (length) {
+            var result = '';
+            var characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
+            var charactersLength = characters.length;
+            for (var i = 0; i < length; i++) {
+                result += characters.charAt(Math.floor(Math.random() * charactersLength));
+            }
+            return result;
+        };
+
         if (!req.body.displayName || !req.body.email || !req.body.password) {
             return res.sendStatus(400);
         }
-
-        permissions.check(res.locals.session.user, {
-            type: "GROUP_ORGANIZE",
-            value: req.body.organizer
-        }).then(function (result) {
-            if (!result) return res.sendStatus(403);
-            req.body.passwordSalt = authHelper.generateSalt(16); // Create salt of 16 characters
-            req.body.passwordHash = authHelper.getPasswordHashSync(req.body.password, req.body.passwordSalt); // Get password hash
-            delete req.body.password; // Delete password permanently
-            log.info({req: req.body}, "EVENTUAL REQUEST");
-            return User.create(req.body).then(function (result) {
-                res.status(201).send(result);
-            }).catch(function (err) {
-                console.error(err);
+        req.body.approvingHash = makeLink(24);
+        req.body.passwordSalt = authHelper.generateSalt(16); // Create salt of 16 characters
+        req.body.passwordHash = authHelper.getPasswordHashSync(req.body.password, req.body.passwordSalt); // Get password hash
+        delete req.body.password; // Delete password permanently
+        req.body.isAdmin = false;
+        log.info({req: req.body}, "EVENTUAL REQUEST");
+        return User.create(req.body).then(function (result) {
+            nodemailer.createTestAccount().then(function () {
+                let transporter = nodemailer.createTransport({
+                    service: 'gmail',
+                    type: "SMTP",
+                    host: "smtp.gmail.com",
+                    secure: true,
+                    // Never fill this data in and add it to git! Only filled in locally or on the server!
```

Comment:
Good addition, maybe explicitly mention password, or even just import it in the startscript of the server